tests/test_pipeline/test_pipeline.py

```diff
@@ -356,3 +362,130 @@ def test_forecast_pipeline_with_nan_at_the_end(df_with_nans_in_tails):
+        assert mask.last_train_timestamp == expected_mask.last_train_timestamp
+        assert [timestamp for timestamp in mask.target_timestamps] == [
+            timestamp for timestamp in expected_mask.target_timestamps
+        ]
+
+
+@pytest.mark.parametrize(
+    "mask", (FoldMask("2020-01-01", "2020-01-02", ["2020-01-03"]), FoldMask("2020-01-03", "2020-01-05", ["2020-01-06"]))
+)
+def test_generate_folds_datasets(simple_ts: TSDataset, mask):
+    """Check _generate_folds_datasets for correct work."""
+    pipeline = Pipeline(model=NaiveModel(lag=7))
+    mask = pipeline._prepare_fold_masks(ts=simple_ts, masks=[mask], mode="constant")[0]
+    train, test = list(pipeline._generate_folds_datasets(simple_ts, [mask], 4))[0]
+    assert train.index.min() == np.datetime64(mask.first_train_timestamp)
+    assert train.index.max() == np.datetime64(mask.last_train_timestamp)
+    assert test.index.min() == np.datetime64(mask.last_train_timestamp) + np.timedelta64(1, "D")
+    assert test.index.max() == np.datetime64(mask.last_train_timestamp) + np.timedelta64(4, "D")
+
+
+@pytest.mark.parametrize(
+    "mask", (FoldMask(None, "2020-01-02", ["2020-01-03"]), FoldMask(None, "2020-01-05", ["2020-01-06"]))
+)
+def test_generate_folds_datasets_without_first_date(simple_ts: TSDataset, mask):
+    """Check _generate_folds_datasets for correct work without first date."""
+    pipeline = Pipeline(model=NaiveModel(lag=7))
+    mask = pipeline._prepare_fold_masks(ts=simple_ts, masks=[mask], mode="constant")[0]
+    train, test = list(pipeline._generate_folds_datasets(simple_ts, [mask], 4))[0]
+    assert train.index.min() == np.datetime64(simple_ts.index.min())
+    assert train.index.max() == np.datetime64(mask.last_train_timestamp)
+    assert test.index.min() == np.datetime64(mask.last_train_timestamp) + np.timedelta64(1, "D")
+    assert test.index.max() == np.datetime64(mask.last_train_timestamp) + np.timedelta64(4, "D")
+
+
+@pytest.mark.parametrize(
+    "lag,expected", ((5, {"segment_0": 76.923077, "segment_1": 90.909091}), (6, {"segment_0": 100, "segment_1": 120}))
+)
+def test_backtest(simple_ts: TSDataset, lag: int, expected: Dict[str, List[float]]):
+    mask = FoldMask(
+        simple_ts.index.min(),
+        simple_ts.index.min() + np.timedelta64(6, "D"),
+        [simple_ts.index.min() + np.timedelta64(8, "D")],
+    )
+    pipeline = Pipeline(model=NaiveModel(lag=lag), transforms=[], horizon=2)
+    metrics_df, _, _ = pipeline.backtest(ts=simple_ts, metrics=[SMAPE()], n_folds=[mask], aggregate_metrics=True)
+    metrics = dict(metrics_df.values)
+    for segment in expected.keys():
+        assert segment in metrics.keys()
+        np.testing.assert_array_almost_equal(expected[segment], metrics[segment])
+
+
+@pytest.mark.parametrize(
+    "lag,expected", ((4, {"segment_0": 0, "segment_1": 0}), (7, {"segment_0": 0, "segment_1": 0.5}))
+)
+def test_backtest_right_timestamps(masked_ts: TSDataset, lag: int, expected: Dict[str, List[float]]):
```

Comment:
What is the difference with previous test?

Reply:
I have no guess) The only difference in the parameters here

Reply:
May be we should change the name of this task somehow.
For example, previous will be `test_backtest_one_point`, the second `test_backtest_two_points`.
Or we should just ask the creator of this tests.

Reply:
I renamed the tests